addons/mass_mailing/models/mailing.py

```diff
@@ -260,15 +260,30 @@ def _compute_statistics(self):
             self.browse(row.pop('mailing_id')).update(row)
 
     def _compute_next_departure(self):
-        cron_next_call = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo().nextcall
+        cron_job = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo()
         str2dt = fields.Datetime.from_string
-        cron_time = str2dt(cron_next_call)
+        cron_time = str2dt(cron_job.nextcall)
+        cron_id = cron_job.id
+
+        # The cron worker usually clean up old cron.triggers after finishing their
+        # jobs, i.e older than now() thus fetched recordset should not be to large
+        now_datetime_buffer = fields.Datetime.now() + relativedelta(minutes=1)
+        now_datetime_str = fields.Datetime.to_string(now_datetime_buffer)
```

Comment:
I believe the ORM will do this for you?

Reply:
(Thanks for the overall input, will comment as I go along)

That was my understanding, but since the original method definition used the `to_string` approach, I didn't know if I should keep the same approach or try to let the ORM do the work. Will try to clean it up a bit.